full-stack-tests/core/src/frontend/standalone/CatalogConnection.test.ts

```diff
@@ -0,0 +1,192 @@
+
+  // Make sure you can create containers for CatalogIModels, create new versions, and read their contents
+  describe("CatalogConnection", async () => {
+    const iTwinId = Guid.createValue();
+
+    before(async () => {
+      await TestUtility.startFrontend(undefined, true);
+      // this test uses the AzTest framework on the backend, and requires its authorization client.
+      await coreFullStackTestIpc.useAzTestAuthClient();
+    });
+
+    after(async () => {
+      // restore the backend's authorization client to its value before this test started
+      await coreFullStackTestIpc.restoreAuthClient();
+      await TestUtility.shutdownFrontend();
+    });
+
+    it("Create container and versions of CatalogIModels within it", async () => {
+      // This file will be used as the "seed catalog" file
+      const localCatalogFile = path.join(process.env.IMODELJS_CORE_DIRNAME!, "core/backend/lib/cjs/test/assets/test.bim");
+
+      // for names of individuals in various apis
+      const people = {
+        fred: "Fred Smith",
+        bill: "Bill Jones",
+        harold: "Harold J. Kennedy",
+        sarah: "Sarah Wilson"
+      } as const;
+
+      // Every cloud container for Catalogs has metadata that can be retrieved from the BlobContainer service
+      // without opening a CatalogIModel. This metadata applies to all versions of the CatalogIModel held in the container.
+      const metadata = { label: "PartsCatalog1", description: "catalog for all projects for ClientA" };
+      // Every CatalogIModel has a "manifest" that describes its purpose. This manifest is versioned inside
+      // the CatalogIModel, so it may be different for each version within a container.
+      const manifest: CatalogIModel.Manifest = {
+        catalogName: "Catalog of Parts",
+        contactName: people.fred,
+        description: "collection of part definitions",
+      };
+
+      // verify that only users with administrator privilege may create new containers
+      await coreFullStackTestIpc.setAzTestUser("readWrite");
+      await expect(CatalogConnection.createNewContainer({ localCatalogFile, version: "1.0.0", iTwinId, manifest, metadata })).rejectedWith("only admins may create containers");
+
+      await coreFullStackTestIpc.setAzTestUser("admin");
+      // verify that illegal dbNames are rejected
+      await expect(CatalogConnection.createNewContainer({ localCatalogFile, dbName: "a:b", version: "1.0.0", iTwinId, manifest, metadata })).rejectedWith("invalid dbName");
+      // verify that an illegal initial version is rejected
+      await expect(CatalogConnection.createNewContainer({ localCatalogFile, version: "not a version", iTwinId, manifest, metadata })).rejectedWith("invalid version specification");
+
+      // create a container for our tests. The initial version is supplied as "1.0.0"
+      const newContainer = await CatalogConnection.createNewContainer({ localCatalogFile, version: "1.0.0", iTwinId, manifest, metadata });
```

Comment:
Will this create a similar problem to workspaceDbs where now v1.0.0 is both empty and immutable?

Reply:
When you call `createNewContainer` to create a new container for a catalog, you upload the initial version of it. That doesn't have to be empty. 

Then, you can supply the initial version number. The default is 0.0.0, which is a "prerelease" version. Like WorkspacesDbs, prerelease versions *are* editable.

This test supplies 1.0.0 to verify that you can override the default.

Reply:
ok cool